test_app/controlledframe_api.js

```diff
@@ -996,19 +996,31 @@ class ControlledFrameController {
 
     let createProperties = {
       checked: $('#context_menus_create_properties_checked_chk').checked,
-      contexts: contexts,
       enabled: $('#context_menus_create_properties_enabled_chk').checked,
-      id: $('#context_menus_create_properties_id_in').value,
-      parentId: $('#context_menus_create_properties_parent_id_in').value,
-      title: $('#context_menus_create_properties_title_in').value,
-      type: $('#context_menus_create_properties_type_in').value,
       onclick: info => {
         let infoJSON = JSON.stringify(info);
         Log.info(`context menu item clicked: ${infoJSON}`);
         $('#context_menus_on_click_result').innerText = infoJSON;
       },
     };
 
+    let id = $('#context_menus_create_properties_id_in').value;
+    if (id.length > 0) {
+      createProperties.id = id;
+    }
```

Comment:
Assuming `context_menus_create_properties_parent_id_in` in index.html becomes `context_menus_create_properties_parentId_in`, then I propose this block of ~16 lines becomes:

```
function setIfValid(object, keyName, keyValue) {
  if (keyValue && keyValue.length > 0) {
    object[keyName] = keyValue;
  }
}

for (const keyName of ["id", "parentId", "title", "type"]) {
  const keyValue = $(`#context_menus_create_properties_${keyName}_in`).value;
  setIfValid(createProperties, keyName, keyValue);
}
```

WDYT?

Reply:
Thanks Chase, this is a much cleaner solution!